Mathlib/AlgebraicTopology/SimplicialSet/Coskeletal.lean

```diff
@@ -0,0 +1,399 @@
+
+`StrictSegal.cosk2Iso : X ≅ (Truncated.cosk 2).obj ((truncation 2).obj X)`
+
+In particular, nerves of categories are 2-coskeletal.
+-/
+
+universe v u
+
+open CategoryTheory Simplicial SimplexCategory Opposite Category Functor Limits
+
+namespace SSet
+
+/-- The identity natural transformation exhibits a simplicial set as a right extension of its
+restriction along `(Truncated.inclusion (n := n)).op`.-/
+@[simps!]
+def rightExtensionInclusion (X : SSet.{u}) (n : ℕ) :
+    RightExtension (Truncated.inclusion (n := n)).op
+      (Functor.op Truncated.inclusion ⋙ X) := RightExtension.mk _ (𝟙 _)
+
+section
+
+local notation (priority := high) "[" n "]" => SimplexCategory.mk n
+
+set_option quotPrecheck false
+local macro:max (priority := high) "[" n:term "]₂" : term =>
+  `((⟨SimplexCategory.mk $n, by decide⟩ : SimplexCategory.Truncated 2))
+
+/-- The object of `StructuredArrow (op [n]) (Truncated.inclusion (n := 2)).op` corresponding to the
+map [0] ⟶ [n] with image `i`. -/
+private
+def pt {n} (i : Fin (n + 1)) : StructuredArrow (op [n]) (Truncated.inclusion (n := 2)).op :=
+  .mk (Y := op [0]₂) (.op (SimplexCategory.const _ _ i))
+
+/-- The object of `StructuredArrow (op [n]) (Truncated.inclusion (n := 2)).op` corresponding to
+the map `[1] ⟶ [n]` with image `i ⟶ i+1`. -/
+private
+def ar {n} (i : Fin n) : StructuredArrow (op [n]) (Truncated.inclusion (n := 2)).op :=
+  .mk (Y := op [1]₂) (.op (mkOfLe _ _ (Fin.castSucc_le_succ i)))
+
+/-- The object of StructuredArrow (op [n]) (Truncated.inclusion (n := 2)).op corresponding to
+`ar k`. -/
+private
+def ar' {n} {i j : Fin (n+1)} (k : i ⟶ j) :
+    StructuredArrow (op [n]) (Truncated.inclusion (n := 2)).op :=
+  .mk (Y := op [1]₂) (mkOfLe _ _ k.le).op
+
+/-- An arrow in `StructuredArrow (op [n]) (Truncated.inclusion (n := 2)).op` arising from
+`const 0 : [0] ⟶ [1]`. -/
+private
+def ar.src {n} (i : Fin n) : (ar i) ⟶ (pt i.castSucc) := by
+  refine StructuredArrow.homMk (.op (SimplexCategory.const _ _ 0)) ?_
+  apply Quiver.Hom.unop_inj
+  ext z; revert z
+  intro (0 : Fin 1)
+  unfold ar pt
+  simp only [StructuredArrow.mk_left, const_obj_obj, len_mk, StructuredArrow.mk_right, op_obj,
+    StructuredArrow.mk_hom_eq_self, Fin.isValue, op_map, Quiver.Hom.unop_op, unop_comp,
+    comp_toOrderHom, OrderHom.comp_coe, Function.comp_apply]
+  rfl
+
+/-- An arrow in `StructuredArrow (op [n]) (Truncated.inclusion (n := 2)).op` arising from
+`const 1 : [0] ⟶ [1]`. -/
+private
+def ar.tgt {n} (i : Fin n) : (ar i) ⟶ (pt i.succ) := by
+  refine StructuredArrow.homMk (.op (SimplexCategory.const _ _ 1)) ?_
+  apply Quiver.Hom.unop_inj
+  ext z; revert z
+  intro (0 : Fin 1)
+  unfold ar pt
+  simp only [StructuredArrow.mk_left, const_obj_obj, len_mk, StructuredArrow.mk_right, op_obj,
+    StructuredArrow.mk_hom_eq_self, Fin.isValue, op_map, Quiver.Hom.unop_op, unop_comp,
+    comp_toOrderHom, OrderHom.comp_coe, Function.comp_apply]
+  rfl
+
+private
+theorem ran.lift.arrow_src {X : SSet.{u}} {n} {i : Fin n}
```

Comment:
What does `ran` refer to in this context? Is there some right Kan extension that isn't syntactically explicit in the type?

Reply:
Moved. Will try to improve comments in the new PR.